consensus/validation_fuzz.go

```diff
@@ -0,0 +1,67 @@
+// +build gofuzz
+
+package consensus
+
+import (
+	"crypto/ed25519"
+	"go.sia.tech/core/types"
+)
+
+func Fuzz(data []byte) int {
+	d := types.NewBufDecoder(data)
+
+	var txn types.Transaction
+	txn.DecodeFrom(d)
+	if err := d.Err(); err != nil {
+		return -1
+	}
+	txn.FileContractResolutions = nil
+
+	var setupTxn types.Transaction
+	for _, sci := range txn.SiacoinInputs {
+		setupTxn.SiacoinOutputs = append(setupTxn.SiacoinOutputs, sci.Parent.SiacoinOutput)
+	}
+	for _, sfi := range txn.SiafundInputs {
+		setupTxn.SiafundOutputs = append(setupTxn.SiafundOutputs, sfi.Parent.SiafundOutput)
+	}
+	for _, fcr := range txn.FileContractRevisions {
+		setupTxn.FileContracts = append(setupTxn.FileContracts, fcr.Parent.FileContract)
+	}
+
+	var ctx ValidationContext
+	au := ApplyBlock(ctx, types.Block{
+		Transactions: []types.Transaction{setupTxn},
+	})
+
+	// use first 32 bytes of transaction data as seed for keys to make it deterministic
+	privkey := ed25519.NewKeyFromSeed(data[:32])
+	pubkey := privkey.Public().([ed25519.PublicKeySize]byte)
```

Comment:
The underlying type here is `[]byte`, not an array. (In case you're wondering, Go crypto APIs avoid arrays because they are copied by value; don't want multiple copies of key material floating around in memory.)